include/deal.II/base/point.h

```diff
@@ -357,6 +370,25 @@ inline Point<dim, Number>::Point(const Number x, const Number y, const Number z)
   this->values[z_index]          = z;
 }
 
+#  ifdef DEAL_II_WITH_BOOST
+template <int dim, typename Number>
+inline Point<dim, Number>::Point(
+  const bg::model::point<Number, dim, bg::cs::cartesian> boost_pt)
+{
+  this->values[0]                = boost::geometry::get<0>(boost_pt);
+  constexpr unsigned int y_index = (spacedim < 2) ? 0 : 1;
+  constexpr unsigned int z_index = (spacedim < 3) ? 0 : 2;
+  if (y_index != 0)
```

Comment:
can you replace this by the original condition `spacedim >= 2` for easier readability? You can then move the declaration of `y_index` into the one use place